java/server/src/org/openqa/grid/internal/utils/SelfRegisteringRemote.java

```diff
@@ -283,10 +283,6 @@ private JsonObject getHubConfiguration() throws Exception{
     String url = api.toExternalForm();
     BasicHttpEntityEnclosingRequest r = new BasicHttpEntityEnclosingRequest("GET", url);
 
-    JsonObject j = new JsonObject();
```

Comment:
Ideally, there should be a check in HTTP object to throw exception if the method is GET and user tries to add payload (I've seen this check in OS source codes). Same can be added as a UT for the method that constructs the HTTP packet, which resides in apache code.